typescript/src/deposit-sweep.ts

```diff
@@ -130,238 +150,297 @@ export async function assembleDepositSweepTransaction(
-    )
-    if (!utxoWithDeposit) {
-      throw new Error("Unknown input")
-    }
+    const utxo = utxos[depositIndex]
+    const previousOutput = Transaction.fromHex(utxo.transactionHex).outs[
+      utxo.outputIndex
+    ]
+    const previousOutputValue = previousOutput.value
+    const previousOutputScript = previousOutput.script
+
+    const deposit = deposits[depositIndex]
 
-    if (previousScript.isScripthash()) {
+    if (isP2SH(previousOutputScript)) {
       // P2SH (deposit UTXO)
-      await signP2SHDepositInput(transaction, i, utxoWithDeposit, walletKeyRing)
-    } else if (previousScript.isWitnessScripthash()) {
+      await signP2SHDepositInput(
+        transaction,
+        inputIndex,
+        deposit,
+        previousOutputValue,
+        keyPair
+      )
+    } else if (isP2WSH(previousOutputScript)) {
       // P2WSH (deposit UTXO)
       await signP2WSHDepositInput(
         transaction,
-        i,
-        utxoWithDeposit,
-        walletKeyRing
+        inputIndex,
+        deposit,
+        previousOutputValue,
+        keyPair
       )
     } else {
       throw new Error("Unsupported UTXO script type")
     }
   }
 
-  const transactionHash = TransactionHash.from(transaction.txid())
+  const transactionHash = TransactionHash.from(transaction.getId())
 
   return {
     transactionHash,
     newMainUtxo: {
       transactionHash,
       outputIndex: 0, // There is only one output.
-      value: BigNumber.from(transaction.outputs[0].value),
+      value: BigNumber.from(transaction.outs[0].value),
     },
     rawTransaction: {
-      transactionHex: transaction.toRaw().toString("hex"),
+      transactionHex: transaction.toHex(),
     },
   }
 }
 
 /**
- * Creates script for the transaction input at the given index and signs the
- * input.
- * @param transaction - Mutable transaction containing the input to be signed.
- * @param inputIndex - Index that points to the input to be signed.
- * @param walletKeyRing - Key ring created using the wallet's private key.
- * @returns Empty promise.
+ * Signs the main UTXO transaction input and sets the appropriate script or
+ * witness data.
+ * @param transaction - The transaction containing the input to be signed.
+ * @param inputIndex - Index pointing to the input within the transaction.
+ * @param prevOutScript - The previous output script for the input.
+ * @param prevOutValue - The value from the previous transaction output.
+ * @param keyPair - A Signer object with the public and private key pair.
+ * @param network - The Bitcoin network type (mainnet or testnet).
+ * @returns An empty promise upon successful signing.
+ * @throws Error if the UTXO doesn't belong to the wallet, or if the script
+ *         format is invalid or unknown.
  */
 async function signMainUtxoInput(
-  transaction: any,
+  transaction: Transaction,
   inputIndex: number,
-  walletKeyRing: any
+  prevOutScript: Buffer,
+  prevOutValue: number,
+  keyPair: Signer,
```

Comment:
`keyPair` -> `walletKeyPair`?

Reply:
Done in 59fb549db1ebf7f92310cd5c1ac05dbdf9f34630.